genestack_client/settings/genestack_config.py

```diff
@@ -102,13 +105,20 @@ def get_text(parent, tag):
                 host = get_text(user, 'host')
                 email = get_text(user, 'email')
                 password = get_text(user, 'password')
+                token = get_text(user, 'token')
                 if not password:
                     try:
                         import keyring
                         password = keyring.get_password(_PASSWORD_KEYRING, alias)
                     except Exception as e:
                         sys.stderr.write('Fail to load password for alias "%s": %s\n' % (alias, e))
-                self.add_user(User(email, alias=alias, host=host, password=password), save=False)
+                if not token:
+                    try:
+                        import keyring
+                        token = keyring.get_password(_TOKEN_KEYRING, alias)
+                    except Exception as e:
+                        sys.stderr.write('Fail to load token for alias "%s": %s\n' % (alias, e))
```

Comment:
Wouldn't it complain about every existing user in config (which obviously have no tokens saved anywhere)?

Reply:
`keyring.get_password` will return None if there is no token in storage.   

Try catch  is required if you had keyring, but it was broken (by changing settings or OS update)